iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/compaction/IcebergCompactionService.java

```diff
@@ -63,6 +59,15 @@ public Boolean compact(Table table, CompactionInfo ci) throws Exception {
       return false;
     }
 
+    if (ci.type == CompactionType.SMART_OPTIMIZE) {
+      ci.type = compactionEvaluator.determineCompactionType();
```

Comment:
Can we create a local var `CompactionType type = ompactionEvaluator.determineCompactionType()` and assign to `ci.type` after the null check?

Reply:
Done